### PR TITLE
Enable HTTP RPC sockets

### PR DESCRIPTION
**Pull request by hiddentao**
This incorporates and replaces ethereum/mist#420. 

* It replaces the `--ipcpath` CLI parameter with `--rpc`, allowing for an HTTP RPC hostname (e.g. `http://localhost:8545`) to be provided. If omitted then IPC is used as the default.
* README updated with instructions
* Introduce usage of `--node-` prefix for CLI parameters that are to be passed onto geth/eth.
* Warns user that the HTTP option is less secure than IPC
* Fix a small bug in solidity compilation error reporting
* Incorporates ethereum/mist#841 as I needed that to test this properly.

### CODE REVIEW

**Comment by hiddentao** [comment]
Ready for review.

**Comment by hiddentao** [comment]
Todo:

* Add warning pop-up

**Comment by hiddentao** [comment]
Dialog msg added.

screen shot 2016-06-28 at 12 25 58

**Comment by frozeman** [comment]
haha :)
Some text improvements:

"....This is less secure than using a local IPC socket - your account passwords will be send in plain text!

Only do this when you secured your HTTP connection, or you know what you are doing."

**Comment by luclu** [comment]
@hiddentao could you also correct [this](https://github.com/ethereum/mist/blob/4922876722fcae21732ed634acb87beccbbaf232/modules/settings.js#L140) typo?

**Comment by hiddentao** [comment]
@luclu Fixed directly in `develop` branch.

File: modules/settings.js [hunk 1]
```diff
@@ -136,8 +137,13 @@ for (let optIdx in argv) {
 }
 
 // some options are shared
+<<<<<<< HEAD
```
Comment (frozeman) [inline modules/settings.js]: This should be removed

**Comment by frozeman** [comment]
There seems to be an issue, if i put `web3.eth.accounts` into the interface it doesn't return, and mist wont quit.
Then sometimes when quitting, it will show all the RPC messages, because it can't parse it properly.

I will look more deeply what could be the problem, but if you have any idea @hiddentao please let me know.

File: modules/settings.js (hunk 1 above)
Comment (hiddentao) [inline modules/settings.js]: Done.

**Comment by hiddentao** [comment]
@frozeman Yep, haven't been able to reproduce it yet. If I see it happen I'll investigate further.

**Comment by hiddentao** [comment]
This is ready to be merged for 0.8.2.

**Pull request merged**